### PR TITLE
Add support for the reMarkable2

### PR DESCRIPTION
**Pull request by alistair23**
This adds support for booting Linux and userspace on the rM2.

### CODE REVIEW

File: documentation/remarkable2.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+SoC.
+
+This layer adds support for building the official kernel source and u-boot
+source for the rM2.
+
+In order to build the images for the rM2 setup this layer and all dependencies
+and then run:
+
+```shell
+MACHINE = "imx7d-remarkable2" bitbake core-image-minimal -k
```
Comment (otavio) [inline documentation/remarkable2.md:13]: I am not against to add the doc, but it ought to match the machine name.

File: recipes-bsp/u-boot/u-boot-remarkable2_2018.03.bb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+LICENSE = "GPLv2+"
+LIC_FILES_CHKSUM = "file://Licenses/README;md5=a2c678cfd4a4d97135585cad908541c6"
+
+require recipes-bsp/u-boot/u-boot.inc
+
+PV = "2018.03+git${SRCPV}"
+
+SRCREV = "47c91918aa7724c16b6eaf87cf5dfbf4548eafc1"
+SRCBRANCH = "zero-sugar"
+SRC_URI = "git://github.com/reMarkable/uboot.git;branch=${SRCBRANCH} \
```
Comment (otavio) [inline recipes-bsp/u-boot/u-boot-remarkable2_2018.03.bb:11]: Have you checked if the U-Boot mainline offers support for it? If so, the patches could be upstreamed and we might use u-boot-fslc.

**Review by otavio — CHANGES_REQUESTED** [review]

File: documentation/remarkable2.md (hunk 1 above)
Comment (alistair23) [reply documentation/remarkable2.md:13]: Fixed.

File: recipes-bsp/u-boot/u-boot-remarkable2_2018.03.bb (hunk 2 above)
Comment (alistair23) [reply recipes-bsp/u-boot/u-boot-remarkable2_2018.03.bb:11]: I haven't checked yet. I'm currently working on upstreaming Linux, after that I can look at u-boot. They use u-boot to do some customised setup so I don't think mainline u-boot would work out of the box.

Comment (otavio) [reply recipes-bsp/u-boot/u-boot-remarkable2_2018.03.bb:11]: Ok; we could apply the patches to linux-fslc in this case.

Comment (alistair23) [reply recipes-bsp/u-boot/u-boot-remarkable2_2018.03.bb:11]: Do you mean `u-boot-fscl`?

There are more then a few patches, I think that will be difficult to do. My hope is just to upstream the diff and then use mainline. I'm hoping to be able to swap to a mainline kernel + patches soon-ish.

**Review by otavio — CHANGES_REQUESTED** [review]
Would you mind renaming the machine to `imx7d-remarkable2`? I've been trying to use the SoC prefix as it helps to know which SoC it uses and also to group the machines.

**Comment by alistair23** [comment]
The machine is called `imx7d-remarkable2`, is there something else I should change?

**Comment by otavio** [comment]
Use this for the doc as well.

**Comment by alistair23** [comment]
Yep! The doc is also called `imx7d-remarkable2`, it's just linux and u-boot that use `remarkable2` in the filename.

**Review by otavio — APPROVED** [review]

**Pull request merged**